src/dune/lib_rules.ml

```diff
@@ -133,7 +137,7 @@ let ocamlmklib ~loc ~c_library_flags ~sctx ~dir ~expander ~o_files ~archive_name
            else
              Command.Args.empty )
          ; A "-o"
-         ; Path (Path.build (Path.Build.relative dir archive_name))
+         ; Path (Path.build (Foreign.Archive.Name.path ~dir archive_name))
```

Comment:
Unrelated to this commit: shouldn't this be `Dep`?

Reply:
No, this is the name of the output archive, see the `-o` argument before. We declare the actual targets separately (there may be more than one: `lib*` and `dll*`).